README.md

```diff
@@ -9,3 +9,152 @@
+
+# Move the binary in to your PATH
+sudo mv "./gardenctl_v2_${os}_${arch}" /usr/local/bin/gardenctl
+```
+
+## Configure Gardenctl
+
+`gardenctl` requires a configuration file. The default location is in `~/.garden/gardenctl-v2.yaml`.
+
+You can modify this file directly using the `gardenctl config` command. It allows adding, modifying and deleting gardens.
+
+Example `config` command:
+``` bash
+# Adapt the path to your kubeconfig file for the garden cluster
+export KUBECONFIG=~/path/to/garden-cluster/kubeconfig.yaml
+
+# Fetch cluster-identity of garden cluster
+CLUSTER_IDENTITY=$(kubectl -n kube-system get configmap cluster-identity -ojsonpath={.data.cluster-identity})
+
+# Configure garden cluster
+gardenctl config set-garden $CLUSTER_IDENTITY --kubeconfig $KUBECONFIG
+```
+This command will create or update a garden with the provided identity and kubeconfig path of your garden cluster.
+
+### Example Config
+
+```yaml
+gardens:
+- identity: landscape-dev # Unique identity of the garden cluster. See cluster-identity ConfigMap in kube-system namespace of the garden cluster
+  kubeconfig: ~/path/to/garden-cluster/kubeconfig.yaml
+# context: different-context # Overrides the current-context of the garden cluster kubeconfig  
+# patterns: ~ # List of regex patterns for pattern targeting
+```
+
+Note: You need to have [gardenlogin](https://github.com/gardener/gardenlogin) installed as `kubectl` plugin in order to use the `kubeconfig`s for `Shoot` clusters provided by `gardenctl`.
+
+### Config Path Overwrite
+
+- The `gardenctl` config path can be overwritten with the environment variable `GCTL_HOME`.
+- The `gardenctl` config name can be overwritten with the environment variable `GCTL_CONFIG_NAME`.
+
+```bash
+export GCTL_HOME=/alternate/garden/config/dir
+export GCTL_CONFIG_NAME=myconfig # without extension!
+# config is expected to be under /alternate/garden/config/dir/myconfig.yaml
+```
+
+### GCTL_SESSION_ID
+
+The state of gardenctl is bound to a shell session and is not shared across windows, tabs or panes.
+A shell session is defined by the environment variable GCTL_SESSION_ID. If this is not defined,
+the value of the TERM_SESSION_ID environment variable is used instead. If both are not defined,
+this leads to an error and gardenctl cannot be executed. The target.yaml and temporary
+kubeconfig.*.yaml files are store in the following directory ${TMPDIR}/garden/${GCTL_SESSION_ID}.
+
+You can make sure that GCTL_SESSION_ID or TERM_SESSION_ID is always present by adding
+the following code to your terminal profile ~/.profile, ~/.bashrc or comparable file.
+```
+bash and zsh: [ -n "$GCTL_SESSION_ID" ] || [ -n "$TERM_SESSION_ID" ] || export GCTL_SESSION_ID=$(uuidgen)
+fish:         [ -n "$GCTL_SESSION_ID" ] || [ -n "$TERM_SESSION_ID" ] || set -gx GCTL_SESSION_ID (uuidgen)
+powershell:   if ( !(Test-Path Env:GCTL_SESSION_ID) -and !(Test-Path Env:TERM_SESSION_ID) ) { $Env:GCTL_SESSION_ID = [guid]::NewGuid().ToString() }
```

Comment:
```suggestion
#### bash and zsh:
```sh
[ -n "$GCTL_SESSION_ID" ] || [ -n "$TERM_SESSION_ID" ] || export GCTL_SESSION_ID=$(uuidgen)
```
#### fish:
```fish
[ -n "$GCTL_SESSION_ID" ] || [ -n "$TERM_SESSION_ID" ] || set -gx GCTL_SESSION_ID (uuidgen)
```
#### powershell:
```pwsh
if ( !(Test-Path Env:GCTL_SESSION_ID) -and !(Test-Path Env:TERM_SESSION_ID) ) { $Env:GCTL_SESSION_ID = [guid]::NewGuid().ToString() }
```
```